### ISSUE TITLE
Site Editor: Consider placing theme blocks higher in the Inserter 

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#42480]
## What problem does this address?

As part of the fifteenth call for testing for the FSE Outreach program, [the following feedback came](https://make.wordpress.org/test/2022/07/11/fse-program-testing-call-15-category-customization/#comment-2608) in from @paaljoachim: 

>Well I went ahead and clicked the blue + button to add a pattern. The Blocks tab screen is first seen. Text blocks are seen in the top. Should I scroll downward to see the various block categories? Should I begin writing in head… into the search box? Perhaps Theme category should be seen at the top since I am in the site editor? As Theme blocks are the more general blocks one might want to add first into a template.

Here's a video of scrolling to find theme blocks if you don't search/don't know what to search for (likely the case for new users):

https://user-images.githubusercontent.com/26996883/179371484-26077bb4-d997-4c50-b777-9bb143b89872.mov

## What is your proposed solution?

Consider placing the theme block category higher in the Inserter when one is in the Site Editor specifically, since it's most likely one will want to use those blocks at least when first building a site. I could see it making sense before Widgets while in the Site Editor while still prioritizing the most common blocks. @WordPress/gutenberg-design for further thoughts!

**Comment by paaljoachim** [comment]
Thank you for making the issue Anne!

A followup.
The block inserter should place the various sections in the context they are most likely needed.
For a post or page. Yes it feels natural to have the order we have today.
For the site editor we need the more general blocks much more visible for users to see. Meaning showing the Theme section at the top so one does not need to even scroll to see the basic building blocks most needed for a template.

**Comment by jameskoster** [comment]
I think there could be some overlap with https://github.com/WordPress/gutenberg/issues/41379 and the discussion [here](https://github.com/WordPress/gutenberg/discussions/35800).

The _order_ of block categories may be less of an issue if they are all visible when you open the Inserter, like in the first video here: https://github.com/WordPress/gutenberg/issues/41379#issuecomment-1143246788.

Just something to keep in mind!

**Comment by mtias** [comment]
Indeed, this shouldn't be an issue if we go with the pattern category treatment for the blocks section as well.

**Comment by mtias** [comment]
Can we do an evaluation of how the inserter with categories might work here?

**Comment by jameskoster** [comment]
A quick mockup using the existing categories:

<img width="362" alt="Screenshot 2023-01-17 at 10 10 45" src="https://user-images.githubusercontent.com/846565/212870538-11bf37ce-c19e-4078-83f1-a7e826871506.png">

Observations:

1. Common blocks are an extra click away. Some shortcuts (or the "Most used" feature from the post editor) might be nice
2. A list allows for more expression in the categories. "Theme" can be broken down into more useful sections.
3. Does "Widgets" still make sense?
4. An existential thought, do we still need the tabbed navigation (separation of insertable objects by type)? Or would it be better if it were possible to view Text blocks and patterns together? e.g:

<img width="619" alt="Screenshot 2023-01-17 at 10 31 16" src="https://user-images.githubusercontent.com/846565/212876017-981e4e52-a336-4cc6-9745-e4985ac0c585.png">

This could work better for sections like "Headers". There we could display existing template parts using the pattern preview rather than the current block button which is a bit lacklustre in that context.

**Comment by richtabor** [comment]
> Common blocks are an extra click away. Some shortcuts (or the "Most used" feature from the post editor) might be nice

I agree, shortcuts or most used blocks would be helpful — to keep those from being buried. 

> Does "Widgets" still make sense?

I don't think so. 

Another note (assuming we make the inserter the same width as the inspector) is that the Blocks/Patterns/Media/Reusable blocks tabs would get tight as-is:

<img width="636" alt="CleanShot 2023-01-25 at 12 54 00" src="https://user-images.githubusercontent.com/1813435/214643801-ca0987b9-4a09-49a0-84dd-b838cd4065ed.png">
